ci/authn-gcp/run_gcloud.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -ex
+
+# Script that runs a Google SDK container executes a script that deploys
+# a Google function which used to obtain id token with `audience` passed in as
+# a parameter in the function URL request query string.
+# The name of the script to run in the Google SDK container.
+GCLOUD_SCRIPT=$1
+
+main() {
+  echo "run_gcloud.sh with script: $GCLOUD_SCRIPT"
+  validate_pre_requisites || exit 1
+  write_sa_key_to_file || exit 1
+  run_gcloud_container || exit 1
+  echo "-> run_gcloud.sh with script: $GCLOUD_SCRIPT done"
+}
+
+validate_pre_requisites() {
+  echo 'validate_pre_requisites'
+
+  if [ -z "$GCLOUD_SCRIPT" ]; then
+    echo "-> Missing script file name argument"
+    exit 1
+  fi
+
+  if [ -z "$GCP_OWNER_SERVICE_KEY" ]; then
+    echo "-- env var 'GCP_OWNER_SERVICE_KEY' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_OWNER_SERVICE_KEY_FILE" ]; then
+    echo "-- env var 'GCP_OWNER_SERVICE_KEY_FILE' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_FETCH_TOKEN_FUNCTION" ]; then
+    echo "-- env var 'GCP_FETCH_TOKEN_FUNCTION' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_PROJECT" ]; then
+    echo "-- env var 'GCP_PROJECT' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$IDENTITY_TOKEN_FILE" ]; then
+    echo "-- env var 'IDENTITY_TOKEN_FILE' is undefined"
+    exit 1
+  fi
+  echo '-> validate_pre_requisites done'
+}
+
+# Authentication with Google CLI requires the service account key as file on disk.
+# This function writes the key provisioned by summon temporary to disk, the cleanup_function.sh script
+# will delete it in the post/always branch of the stage.
+# The file is used by bith deploy and clean function scripts.
+write_sa_key_to_file() {
+  echo 'write_sa_key_to_file'
+  if [ ! -f "$GCP_OWNER_SERVICE_KEY" ]; then
+    echo "-- Write account service key to file: '$GCP_OWNER_SERVICE_KEY_FILE' (required for gcloud auth)"
+    echo "$GCP_OWNER_SERVICE_KEY" >"$GCP_OWNER_SERVICE_KEY_FILE"
```

Comment:
don't know if it will effect anything but I think `>` should have spaces surrounding it?